server/tapeuse.go

```diff
@@ -0,0 +1,64 @@
+package server
+
+import (
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+// A FixityDB tracks the information the fixity service needs to know what
+
+func (s *RESTServer) EnableTapeUse() {
+	log.Println("Enabling Bendo Tape Use")
+	s.TapeUsed = true
+	s.Items.SetUseStore(true)
+}
+
+func (s *RESTServer) DisableTapeUse() {
+	log.Println("Disabling Bendo Tape Use")
+	s.TapeUsed = false
+	s.Items.SetUseStore(false)
+}
+
+func (s *RESTServer) StatusTapeUse() bool {
+	return s.TapeUsed
+}
+
+// SetTapeUseHandler handles requests to PUT /admin/use_tape/:status
+func (s *RESTServer) SetTapeUseHandler(w http.ResponseWriter, r *http.Request, ps httprouter.Params) {
+	log.Println("Called PUT /admin/user_tape")
+
+	status := ps.ByName("status")
+
+	switch status {
+	case "on":
+		w.WriteHeader(201)
+		// start Enable Process, unless already enabled
+		if s.StatusTapeUse() == false {
+			go s.EnableTapeUse()
+		}
+	case "off":
+		w.WriteHeader(201)
+		// start Disable Process, unless already disabled
+		if s.StatusTapeUse() == true {
+			go s.DisableTapeUse()
+		}
+	default:
+		w.WriteHeader(503)
+		log.Println("PUT /admin/user_tape: unknown parameter")
```

Comment:
Done

On Thu, Aug 25, 2016 at 12:46 PM, Don Brower <notifications@github.com>
wrote:

> In server/tapeuse.go
> <https://github.com/ndlib/bendo/pull/136#discussion_r76280751>:
>
> > +	switch status {
> > +	case "on":
> > +		w.WriteHeader(201)
> > +		// start Enable Process, unless already enabled
> > +		if s.StatusTapeUse() == false {
> > +			go s.EnableTapeUse()
> > +		}
> > +	case "off":
> > +		w.WriteHeader(201)
> > +		// start Disable Process, unless already disabled
> > +		if s.StatusTapeUse() == true {
> > +			go s.DisableTapeUse()
> > +		}
> > +	default:
> > +		w.WriteHeader(503)
> > +		log.Println("PUT /admin/user_tape: unknown parameter")
>
> if you are logging an error, might as well log what the unknown parameter
> is.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/ndlib/bendo/pull/136/files/116b45889c785d5598950eba80d6730d33f98945#r76280751>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADNGTCvgqBGMRk6kwzdRBH00CTXr_bPAks5qjcbrgaJpZM4JsckT>
> .
>